Sources/ClientRuntime/Networking/Http/Middlewares/DeserializeMiddleware.swift

```diff
@@ -29,9 +38,7 @@ public struct DeserializeMiddleware<Output: HttpResponseBinding,
                 // and then the service error eg. [AccountNotFoundException](https://github.com/awslabs/aws-sdk-swift/blob/d1d18eefb7457ed27d416b372573a1f815004eb1/Sources/Services/AWSCloudTrail/models/Models.swift#L62)
                 let bodyData = try await copiedResponse.httpResponse.body.readData()
                 copiedResponse.httpResponse.body = .data(bodyData)
-                let error = try await OutputError.makeError(httpResponse: copiedResponse.httpResponse,
-                                            decoder: decoder)
-                throw error
+                throw try await httpResponseErrorClosure(copiedResponse.httpResponse)
```

Comment:
Same thing for response error as for response output above - the closure does the work of calling the `makeError` method, and the middleware needs to know nothing about the error response type.  So we can clear `HttpResponseErrorBinding` out of every middleware where it is referenced.